### DISCUSSION TITLE
Integration in Nextcloud Cospend

### DISCUSSION THREAD

**Discussion by linux-lukas** [pretix/pretix#2158] [Ideas]
I could imagine that it could give a lot of meaning if pretix had a (optional) interface to the nextcloud app cospend.

**Comment by raphaelm** [comment]
Can you give some conetxt on what that that app does and how such an integration would look like? Can you given an example on how it could make things easier for you?

**Reply by linux-lukas** [reply]
Thank you for your message, @raphaelm. 

The Cospend application for Nextcloud is first of all a budget manager for communities and groups. It can also be used to create project accounts. 

My thought was the following: 

Let's assume that a non-profit organisation uses a Nextcloud instance including Cospend, but uses pretix for cost processing (e.g. ticket management). Let's suppose that the event or series of events is sponsored by the authorities and subsequent proof in the form of a financial project statement is to be submitted to the sponsoring authority. In order to avoid having to import this manually, I could imagine that an integration, in so far as the transacted amounts of money, without further details, only the money and what it was used for, could be imported from pretix to Cospend, would make sense in my opinion. 

Basically, a way to integrate the payment transactions (only the amount of money) into Cospend to make them available for project accounting. 

I hope that I was able to answer your question sufficiently. 

The advantage, as I see it, is that it would be easier for people who are not so technically experienced, for example those who are responsible for finance.

The API of Cospend is documented here: https://github.com/eneiluj/cospend-nc/blob/master/docs/dev.md

**Reply by raphaelm** [reply]
Thanks for elaborating!

This is not something that we will do, but that someone from the community could certainly implement as a pretix plugin! :)

**Reply by linux-lukas** [reply]
> Thanks for elaborating!
> 
> This is not something that we will do, but that someone from the community could certainly implement as a pretix plugin! :)

With pleasure. Would there be a possibility to label this discussion that the idea should come from the community as a plugin?

**Reply by linux-lukas** [reply]
One or more hackerthron's would certainly be suitable for the realisation of such a plugin, wouldn't they @raphaelm ?

**Reply by raphaelm** [reply]
Sorry, I have no idea how much work this would be.
